src/Raven.Server/Documents/StreamsTempFile.cs

```diff
@@ -186,6 +188,20 @@ public IDisposable CreateReaderStream(out LimitedStream limitedStream)
                 });
             }
 
+            public LimitedStream CreateReaderStream()
+            {
+                var streamDispose = CreateReaderStream(out var stream);
+                var disposableAction = new DisposableAction(() =>
```

Comment:
create a dedicated struct for this dispose?

Reply:
done